Engine/source/gfx/gfxTextureArray.h

```diff
@@ -0,0 +1,72 @@
+//-----------------------------------------------------------------------------
+// Copyright (c) 2012 GarageGames, LLC
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//-----------------------------------------------------------------------------
+
+#ifndef _GFXTEXTUREARRAY_H_
+#define _GFXTEXTUREARRAY_H_
+
+#ifndef _REFBASE_H_
+#include "core/util/refBase.h"
+#endif
+#ifndef _GFXRESOURCE_H_
+#include "gfx/gfxResource.h"
+#endif
+#ifndef _GFXENUMS_H_
+#include "gfxEnums.h"
+#endif
+#ifndef _GFXTEXTUREHANDLE_H_
+#include "gfxTextureHandle.h"
+#endif
+
+
+class GFXTextureProfile;
+class GFXTextureObject;
+
+class GFXTextureArray : public StrongRefBase, public GFXResource
+{
+public:
+   virtual bool fromTextureArray(Vector<GFXTexHandle> textureArray) = 0;
```

Comment:
unnecessary copy happening here, pass the vector by reference (preferably const)